algorithms/dp/bitmasking.py

```diff
@@ -1,19 +1,110 @@
+from collections import defaultdict
+
+MAX_NB_PEOPLE = 10
+
+"""
+Check that the given caps sets is valid 
+
+:param capsSets: The caps sets to check
+:param nbCaps: The number of caps
+:returns: Nothing but raises errors if invalid caps sets
+"""
+def checkArgument(capsSets: list, nbCaps: int):
+    if len(capsSets) > MAX_NB_PEOPLE:
+        raise ValueError(f'The maximum number of people is {MAX_NB_PEOPLE}, but you gave {len(capsSets)} people !')
+    for capsSet in capsSets:
+        if not isinstance(capsSet, list):
+            raise ValueError(f'The given caps set {capsSet} is not valid !')
+        for cap in capsSet:
+            if not isinstance(cap, int):
+                raise ValueError(f'The given cap "{cap}" in caps set {capsSet} is not valid !')
+            if cap > nbCaps:
+                raise ValueError(f'The maximum cap id is {nbCaps}, but you gave a cap id "{cap}" in caps set {capsSet} !')
```

Comment:
I kind of think this error handling code distracts from the actual implementation. I think it is not necessary to be this through as the purpose of this implementation is to demonstrate the algorithm. So if a user provides arguments using the wrong type, the program will just blow up with a type error. 

What is your take?

Reply:
I think it would be helpful to keep it, but we can mention that the call of this function is optional and let the user choose if he wants to check or not.

Reply:
Alright I am fine with that